src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForCreateCodeFixProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCollectionExpression;
+
+using static SyntaxFactory;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseCollectionExpressionForCreate), Shared]
+internal partial class CSharpUseCollectionExpressionForCreateCodeFixProvider
+    : ForkingSyntaxEditorBasedCodeFixProvider<InvocationExpressionSyntax>
+{
+    private static readonly SyntaxAnnotation s_dummyObjectAnnotation = new();
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpUseCollectionExpressionForCreateCodeFixProvider()
+        : base(CSharpCodeFixesResources.Use_collection_expression,
+               IDEDiagnosticIds.UseCollectionExpressionForCreateDiagnosticId)
+    {
+    }
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(IDEDiagnosticIds.UseCollectionExpressionForCreateDiagnosticId);
+
+    protected override async Task FixAsync(
+        Document document,
+        SyntaxEditor editor,
+        CodeActionOptionsProvider fallbackOptions,
+        InvocationExpressionSyntax invocationExpression,
+        ImmutableDictionary<string, string?> properties,
+        CancellationToken cancellationToken)
+    {
+        var unwrapArgument = properties.ContainsKey(CSharpUseCollectionExpressionForCreateDiagnosticAnalyzer.UnwrapArgument);
+
+        // We want to replace `XXX.Create(...)` with the new collection expression.  To do this, we go through the
+        // following steps.  First, we replace `XXX.Create(a, b, c)` with `new(a, b, c)` (an dummy object creation
+        // expression). We then call into our helper which replaces expressions with collection expressions.  The reason
+        // for the dummy object creation expression is that it serves as an actual node the rewriting code can attach an
+        // initializer to, by which it can figure out appropriate wrapping and indentation for the collection expression
+        // elements.
```

Comment:
a really funky way to do things.  but it allows us to leverage as much rewriting code as possible.